frontend/src/concepts/analyticsTracking/useTrackUser.ts

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { useUser } from '~/redux/selectors';
+import { useAccessReview } from '~/api';
+import { AccessReviewResourceAttributes } from '~/k8sTypes';
+import { IdentifyEventProperties } from '~/concepts/analyticsTracking/trackingProperties';
+
+export const useTrackUser = (username?: string): IdentifyEventProperties => {
+  const { isAdmin } = useUser();
+  const [anonymousId, setAnonymousId] = React.useState<string | undefined>(undefined);
+
+  const createReviewResource: AccessReviewResourceAttributes = {
+    group: 'project.openshift.io',
+    resource: 'projectrequests',
+    verb: 'create',
+  };
+  const [allowCreate] = useAccessReview(createReviewResource);
+
+  const computeAnonymousUserId = async () => {
+    const anonymousIDBuffer = await crypto.subtle.digest(
+      'SHA-1',
+      new TextEncoder().encode(username),
+    );
+    const anonymousIDArray = Array.from(new Uint8Array(anonymousIDBuffer));
+    const aId = anonymousIDArray.map((b) => b.toString(16).padStart(2, '0')).join('');
+    setAnonymousId(aId);
+  };
+
+  if (!anonymousId) {
+    computeAnonymousUserId();
+  }
+
+  const props: IdentifyEventProperties = {
+    isAdmin,
+    canCreateProjects: allowCreate,
+    anonymousID: anonymousId,
+  };
```

Comment:
If you use `props` as a dependency, then it must be a stable reference as much as possible.
```suggestion
  const props: IdentifyEventProperties = React.useMemo(() =>({
    isAdmin,
    canCreateProjects: allowCreate,
    anonymousID: anonymousId,
  }), [isAdmin, allowCreate, anonymousId];
```